src/main/java/org/ballerinax/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,170 @@
+        }
+        return serialized;
+    }
+
+    @Override
+    public void visit(TypeDefinitionNode typeDefinitionNode) {
+        if (typeDefinitionNode.typeDescriptor().kind() == SyntaxKind.RECORD_TYPE_DESC) {
+            Optional<Symbol> recordSymbolOpt = this.model.symbol(typeDefinitionNode);
+            if (recordSymbolOpt.isPresent()) {
+                Symbol recordSymbol = recordSymbolOpt.get();
+                Optional<String> recordNameOpt = recordSymbol.getName();
+                if (recordNameOpt.isPresent()) {
+                    String recordName = recordNameOpt.get();
+                    String recordSignature = recordSymbol.getModule().get().id().toString();
+                    recordName = recordSignature + ":" + recordName;
+                    Map<String, RecordFieldSymbol> fieldSymbolMap = ((RecordTypeSymbol) ((TypeDefinitionSymbol)
+                            recordSymbol).typeDescriptor()).fieldDescriptors();
+                    String serialized = getFieldTypes(fieldSymbolMap);
+                    serialized = "{\"" + recordName + "\":" + serialized + "}";
+                    this.recordTypes.put(recordName, serialized);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        NodeList<Token> classTypeQualifiers = classDefinitionNode.classTypeQualifiers();
+        for (Token classTypeQualifier : classTypeQualifiers) {
+            if (classTypeQualifier.text().equals("client")) {
+                Optional<Symbol> classSymbolOpt = model.symbol(classDefinitionNode);
+                if (classSymbolOpt.isPresent()) {
+                    Symbol classSymbol = classSymbolOpt.get();
+                    if (classSymbol.getName().isEmpty()) {
+                        continue;
+                    }
+                    String clientName = classSymbol.getName().get();
+                    Collection<MethodSymbol> methods = ((BallerinaClassSymbol) classSymbol).methods().values();
+                    Map<String, FunctionRecord> functionMap = new HashMap<>();
+                    for (MethodSymbol method : methods) {
+                        boolean checkForRemote = false;
+                        if (method.qualifiers().contains(Qualifier.REMOTE)) {
+                            checkForRemote = true;
+                        }
+                        if (checkForRemote) {
+                            String functionName = method.getName().get();
+                            FunctionRecord functionRecord = new FunctionRecord();
+                            for (ParameterSymbol parameter : method.typeDescriptor().parameters()) {
+                                if (parameter.getName().isPresent()) {
+                                    String parameterName = parameter.getName().get();
+                                    String paraType = parameter.typeDescriptor().typeKind().toString();
```

Comment:
Need to get the parameter type matched to the old version

Reply:
Done